sdk/identity/azure-identity/src/azure_cli_credential.cpp

```diff
@@ -171,10 +173,12 @@ class OutputPipe final {
 
 private:
 #if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WINAPI_PARTITION_DESKTOP) || WINAPI_PARTITION_DESKTOP // not UWP
```

Comment:
It's unfortunate we have so many ifdefs to maintain uwp.

Separate from this PR: Based on telemetry, we should really consider dropping support for it all together.

Reply:
Not "for" or "against", but as an Azure SDK developer:
* As of programming environment, "UWP" is pretty whimsy platform to port the code for.
* The fact that on UWP you can have libcurl but not WinHTTP blows my mind.
* The set of macros to check for UWP is a monstrosity, I actually go and inject `#define`s and `#undef`s each time I write a macro condition for UWP, and look if the code being shadowed by syntax highlighting meets my expectations.
* The limitations, such as "you have no getenv(), but GetEnvironmentVariableA() will work" seem random and artificial. Thats getenv() which has been there probably since 80s and which is there on all other platforms.
* All that complexity comes from an API with rather small market share, for which one of the reasons could be the things listed above.

Reply:
FWIW, on UWP you can have WInHTTP, as of relatively recent WIndows builds - for instance, it works just fine on current WIndows SDKs/VS 2022. But when we originally did that, it wasn't supported.

THe core reason for that is that the networking team wanted to get people moved from WInHTTP to Windows.Networking.Http, which is a "better" version of WinHTTP.

Reply:
Btw, apparently the cognitive services folks consider UWP and the WIndows Store an important customer scenario.

Reply:
It is fine, we can do it, it will just be annoying sometimes :)
@LarryOsterman, it is great that it is supported, now the question is how we can find it out in compile-time.
But suppose we can, and we put preprocessor condition in the code: "if not UWP or Windows SDK version >= {recent_version}".
But vcpkg will not be able to see that, at least not that easily. For the "winhttp" feature, vcpkg manifest says that it is supported on regular Windows but not UWP, which is true if you have old Windows SDK installed. But vcpkg has no way to detect which Windows SDK version is installed. At least, not that I know of, and if it can, we'll need to write some complex things to make it work.
Probably our best bet is to enable customers to force-install the feature even on UWP, while by default it will be disabled.

Reply:
We also can refactor transport implementations from being vcpkg features into being a monofeature: as one way to do it, you'd get a factory to create winhttp transport adapter even on linux, but it will fail in runtime. Yes implement that, but also make sure the SDK code is backwards compatible.